insights/core/plugins.py

```diff
@@ -223,16 +240,51 @@ def make_response(error_key, **kwargs):
     }
     kwargs.update(r)
 
-    # using str() avoids many serialization issues and runs in about 75%
-    # of the time as json.dumps
     detail_length = len(str(kwargs))
 
-    if detail_length > settings.defaults["max_detail_length"]:
-        log.error("Length of data in make_response is too long.", extra={
-            "max_detail_length": settings.defaults["max_detail_length"],
-            "error_key": error_key,
-            "len": detail_length
-        })
+    if _is_make_reponse_too_long(error_key, kwargs):
+        r["max_detail_length_error"] = detail_length
+        return r
+
+    return kwargs
+
+
+def make_fingerprint(fingerprint_key, **kwargs):
+    """ Returns a JSON document appropriate as a fingerprint rule plugin final
+    result.
+
+    :param str fingerprint_key: The fingerprint key name is used for
+          identification of the plugin.
+    :param \*\*kwargs: Strings to pass additional information to the frontend for
+          rendering more complete messages in a customer system report.
+
+
+    Given::
+
+        make_fingerprint("FINGERPRINT", manufacturer="Red Hat", product="Insights")
+
+    The response will be the JSON string ::
+
+        {
+            "type": "fingerprint",
+            "fingerprint_key": "FINGERPRINT",
+            "manufacturer": "Red Hat",
+            "product": "Insights"
+        }
+    """
+
+    if "fingerprint_key" in kwargs or "type" in kwargs:
+        raise Exception("Can't use an invalid argument for make_fingerprint")
```

Comment:
@pratikgadiya12 to me consistent with the similar functions shouldn't this be a `ValueError` exception?